artiq/firmware/runtime/main.rs

```diff
@@ -58,6 +59,14 @@ mod session;
 mod moninj;
 #[cfg(has_rtio_analyzer)]
 mod analyzer;
+mod dhcp;
+
+// Fixed indexes for the IP addresses so that they can be modified with some
+// semblance of confidence
```

Comment:
Why not use a Rust struct?

Reply:
These indexes are used for the positions of these IP addresses with in `smoltcp::iface::interface::InterfaceInner::ip_addrs` which is a `ManagedSlice<'a, IpCidr>`. Using a struct unfortunately isn't an option, we would require smoltcp to be generic over the datastructure that it uses to store IP addresses.

Reply:
What about building the list from the struct just before passing it to smoltcp?